src/test/groovy/org/sonarqube/gradle/SonarQubePluginTest.groovy

```diff
@@ -29,688 +31,730 @@ import org.gradle.initialization.GradlePropertiesController
+  }
 
-    def "doesn't add sonar properties for skipped child project when using old sonarqube task"() {
-        childProject2.sonarqube.skipProject = true
+  def "set jdkHome for 'java' projects"() {
+    parentProject.pluginManager.apply(JavaPlugin)
 
-        when:
-        def properties = parentSonarTask().properties.get()
+    when:
+    def properties = parentSonarTask().properties.get()
 
-        then:
-        !properties.any { key, value -> value.contains("child2") }
-    }
+    then:
+    properties["sonar.java.jdkHome"] != null
+  }
 
-    def "doesn't add sonar properties for skipped child projects"() {
-        childProject.sonar.skipProject = true
+  def "set jdkHome for 'groovy' projects"() {
+    parentProject.pluginManager.apply(GroovyPlugin)
 
-        when:
-        def properties = parentSonarTask().properties.get()
+    when:
+    def properties = parentSonarTask().properties.get()
 
-        then:
-        !properties.any { key, value -> key.startsWith("child.sonar.") }
-        !properties.any { key, value -> key.startsWith(":parent:child.") }
-    }
+    then:
+    properties["sonar.java.jdkHome"] != null
+  }
 
-    def "doesn't add sonar properties for skipped projects"() {
-        parentProject.sonar.skipProject = true
+  def JVM_SOURCE_FILE_JAVA = "src/test/projects/kotlin-multiplatform-project/src/jvmMain/java/me/user/application/Sample.java"
+  def JVM_SOURCE_FILE_KOTLIN = "src/test/projects/kotlin-multiplatform-project/src/jvmMain/kotlin/me.user.application/Sample.kt"
+  def JVM_SOURCE_FILE_JS = "src/test/projects/kotlin-multiplatform-project/src/jsMain/kotlin/Sample.js"
 
-        when:
-        def properties = parentSonarTask().properties.get()
+  def "add Kotlin multiplatform sources"() {
+    def jvMainSourceSet = mockKotlinSourceSet("jvmMain", Set.of(new File(JVM_SOURCE_FILE_JAVA), new File(JVM_SOURCE_FILE_KOTLIN)))
+    def jsMainSourceSet = mockKotlinSourceSet("jsMain", Set.of(new File(JVM_SOURCE_FILE_JS)))
 
-        then:
-        !properties.any { key, value -> key.startsWith("sonar.") }
-        !properties.any { key, value -> key.startsWith(":parent.") }
-    }
+    def kotlinSourceSetContainer = mock(NamedDomainObjectContainer<KotlinSourceSet>.class)
+    // return a new stream on each invocation
+    when(kotlinSourceSetContainer.stream()).then(invocation -> Arrays.stream(jvMainSourceSet, jsMainSourceSet))
 
-    def "set jdkHome for 'java' projects"() {
-        parentProject.pluginManager.apply(JavaPlugin)
+    def multiplatformExtension = mock(KotlinMultiplatformExtension.class)
+    when(multiplatformExtension.getSourceSets()).thenReturn(kotlinSourceSetContainer)
 
-        when:
-        def properties = parentSonarTask().properties.get()
+    parentProject.extensions.add("kotlin", multiplatformExtension)
+    // todo replace with KotlinMultiplatformPlugin? How do we get the plugin available for UT?
+    parentProject.pluginManager.apply(JavaPlugin)
```

Comment:
I think we should find a way to apply the KotlinMultiplatformPlugin instead of the JavaPlugin here